Portable mode for windows

ctrlaltca opened the pull request:
## Related Ticket(s)
- Fixes #1616
- Fixes #2789

## Short roundup of the initial problem
Cockatrice already supported portable mode, but had to be compiled with a special flag. Since we didn't provide portable mode builds, this was impossible for users to use.

## What will change with this Pull Request?
Portable mode is now activated when a spacial file named "portable.dat" is present in the same folder of cockatrice.exe.
The NSIS installer offers a choice between normal and portable install; normal installs works as before, while for portable installs there are a few differences:
 * the portable.dat file is created by NSIS during the install;
 * the default install directory is the user's desktop\Cockatrice;
 * no registry keys are touched: no uninstall entry is created, and no check for a previous installation is executed;
 * the "components" dialog is not shown; the "application" is installed and the "start menu" component is not installed.

While running in portable mode, Cockatrice will log "Portable mode enabled" in the debug log and disable the "edit" button for paths selection in the settings dialog.

Additional changes:
 * ask for administrator permissions when installing in "normal" mode;
 * support command line flags (portable mode, silent install, destination dir)
## Screenshots

![schermata 2017-07-03 alle 20 00 34](https://user-images.githubusercontent.com/1631111/27803545-513a0eb6-602a-11e7-887c-423c4f80c3b0.png)

tooomm commented:
This is awesome! 👍 

My observations/questions/suggestions with this are:
- Admin mode still requiered on file start, not just on "normal mode" installation (`Next >`)
Maybe you are aware of this and this is subject to change regarding your part at the end - wasn't quite sure.
- How do I update a portable version? is the internal installer dealing with a portable update correctly?
- Default location should be `desktop\Cockatrice portable` instead with **portable** branding
- The path in settings have a weird double slash `//` for one directory
![port_paths](https://user-images.githubusercontent.com/9874850/27841297-565761f6-6100-11e7-8cdc-eb8976fc03d5.png)
- I think as well that just disabling the edit button feels weird, the user things it's a normal input field and try to click or mark to do changes. It should be visible, but grayed out on top!
- The `Paths` headline for that settings group could additionally extend to: `Paths (editing disabled in portable mode)`
- There is a weird first run interaction if you hit the `settings` button on the dialog which asks you to run Oracle because there is no database found and you can't edit the paths. Cockatrice tries to offer you to change the paths but that doesn't work for sure.
- Phrasing `install` as noun is more colloquial I guess. should be `Portable installation` or `Install in portable mode` or just `Portable mode`. Same for the other option: `install normally` or `Standard installation`/`Normal installation`
Maybe also add a small explanation line like in the foobar example in #1616 
- `data` folder should be something like `user data` for clearness that that's the place where user can/should edit stuff. Normally it's also placed within the "user" folder, so...

ctrlaltca commented:
> Admin mode still requiered on file start, not just on "normal mode" installation (Next >)
> Maybe you are aware of this and this is subject to change regarding your part at the end - wasn't quite sure.

You can't elevate to admin rights in the middle of an application, but only when you start it.
Anyway, if you are an administrator on your pc, it will ask for admin elevation, and will permit you to install both in normal and in portable mode.
If you are not an administrator on your pc, it won't ask for admin elevation, but it will only permit you to install in portable mode:
![schermata 2017-07-05 alle 09 19 25](https://user-images.githubusercontent.com/1631111/27853479-14b8ca7e-6163-11e7-9fce-64317a2db940.png)

> How do I update a portable version? is the internal installer dealing with a portable update correctly?

Re-installing the application over the same directory. I've added a "cleaning" routine in fbab7f0 that triggers when we are installing in portable mode on a directory where a previous portable mode installation is found.

> Default location should be desktop\Cockatrice portable instead with portable branding

Looking at https://portableapps.com/ as an example, they use "NamePortable" without spaces. I'd stick to their habits. Addressed in 60ce787

> The path in settings have a weird double slash // for one directory

This also happens in normal installations, i'll have a look in another PR.

> I think as well that just disabling the edit button feels weird, the user things it's a normal input field and try to click or mark to do changes. It should be visible, but grayed out on top!

Agreed, fixed in da3ca60

> The Paths headline for that settings group could additionally extend to: Paths (editing disabled in portable mode) or Paths (you can't change the locations in portable mode)

Agreed, fixed in da3ca60

> There is a weird first run interaction if you hit the settings button on the dialog which asks you to run Oracle because there is no database found and you can't edit the paths. Cockatrice tries to offer you to change the paths but that doesn't work for sure.

Yep, this is annoying, but i didn't come up with a good solution.. other that disable that button, doesn't really seems a good solution.

> Phrasing install as noun is more colloquial I guess. should be Portable installation or Install in portable mode or just Portable mode. Same for the other option: install normally or Standard installation/Normal installation

Addressed in 7568c27

> data folder should be something like user data for clearness that that's the place where user can/should edit stuff. Normally it's also placed within the "user" folder, so...

Sticking with https://portableapps.com/, they use "Data", and i don't see a good reason to change it.
